### PR TITLE
Point cloud display settings

### LINKED ISSUE #4342: Add display settings for point clouds
**Is your feature request related to a problem? Please describe.**
When rendering point clouds, iTwin.js currently automatically decides the point size. In some cases, it would be useful to be able to scale this size as a user in order to discern various properties of the scene.

**Describe the solution you'd like**
Add an API to scale the size of the points used by the point cloud shader.

**Describe alternatives you've considered**
None.

**Additional context**
TBD.

### PR DESCRIPTION
**Pull request by pmconne**
Fixes #4342.

Add RealityModelDisplaySettings for customizing the display of reality models. These can be associated with a context reality model or a persistent reality model. Each model has its own settings - there are no top-level settings associated with the display style.

The settings contain the previously hard-coded ratio by which we mix override color with reality model color - it can now be configured. The rest of the settings currently apply only to point clouds:
- Size mode: point size can be computed from voxel size (as before), or specified in pixels (new).
- A scale can be applied to voxel size as well as customization of the min and max sizes in this mode.
- Points can be drawn round (as before) or square (new, as in several point clouds viewers including Cesium).

display-test-app contains a new key-in `dta reality model settings` to open a hilariously crude dialog for editing these settings. It currently operates only on the first reality model it finds in the active viewport.

Incidentally, also fix hilite shader failing to produce round points.

TODO
- [x] NextVersion.md
- [x] Unit tests
- [x] Run ImageTests - no differences reported
- [ ] (Post-merge) ImageTests test cases

### CODE REVIEW

File: core/common/src/RealityModelDisplaySettings.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+export class PointCloudDisplaySettings {
+  public readonly shape: PointCloudShape;
+  public readonly sizeMode: PointCloudSizeMode;
+  public readonly pixelSize: number;
+  public readonly voxelScale: number;
+  public readonly minPixelsPerVoxel: number;
+  public readonly maxPixelsPerVoxel: number;
+
```
[the remark this answers is not in the document]
Comment (markschlosseratbentley) [reply core/common/src/RealityModelDisplaySettings.ts:35]: + @DStradley FYI

**Review by DStradley — APPROVED** [review]

File: test-apps/display-test-app/src/frontend/RealityModelDisplaySettingsWidget.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+interface RealityModel {
+  readonly name: string;
+  settings: RealityModelDisplaySettings;
+  appearance: FeatureAppearance | undefined;
+}
+
+class ContextModel implements RealityModel {
+  private readonly _state: ContextRealityModelState;
+
+  public constructor(state: ContextRealityModelState) {
```
Comment (ben-polinsky) [inline test-apps/display-test-app/src/frontend/RealityModelDisplaySettingsWidget.ts:26]: Could use constructor initialization shorthand here/throughout this file - `public constructor(private readonly _state: ContextRealityModelState)`

Comment (pmconne) [reply test-apps/display-test-app/src/frontend/RealityModelDisplaySettingsWidget.ts:26]: I absolutely despise that and delete it wherever I come across it in our code.

Comment (ben-polinsky) [reply test-apps/display-test-app/src/frontend/RealityModelDisplaySettingsWidget.ts:26]: Understood - saw it used in some places in the repo so wasn't sure if there was an opinion on it or not.

**Pull request merged**